js/form/src/DropdownOptions.svelte

```diff
@@ -10,9 +10,30 @@
 	let distance_from_top: number;
 	let distance_from_bottom: number;
 	let input_height: number;
+	let input_width: number;
 	let refElement: HTMLDivElement;
 	let listElement: HTMLUListElement;
 	let top: string | null, bottom: string | null, max_height: number;
+	let innerHeight: number;
+
+	const calculate_window_distance = () => {
+    const {top, bottom} = refElement.getBoundingClientRect(); 
+		distance_from_top = innerHeight - top;
+		distance_from_bottom = innerHeight - bottom;
+	};
```

Comment:
This logic isn't quite right. What we care about here is how close we are to the edge of the viewport. These values are always equal to one another because the height of the element is 0, so their `top` and `bottom` is always the same. `getBoundingClientRect` returns values relative to the top of viewport. We need to do this:

```js
distance_from_top = top;
distance_from_bottom = innerHeight - bottom;
```